docs/source/reference/tpu.rst

```diff
@@ -0,0 +1,183 @@
+
+The above YAML considers :code:`n1-highmem-8` as the host machine and :code:`tpu-v2-8` as the TPU node resource.
+You may modify the host instance type or TPU type as you wish.
+To show more TPU accelerators, you may run the command :code:`sky show-gpus --all` or refer to GCP's `documentation <https://cloud.google.com/tpu/docs/regions-zones>`_.
+
+Now, we show a complete YAML for running `mnist training <https://cloud.google.com/tpu/docs/tutorials/mnist-2.x>`_ on TPU node with TensorFlow.
+
+.. code-block:: yaml
+
+   # Task name (optional), used for display purposes.
+   name: mnist-tpu-node
+
+   resources:
+      accelerators: tpu-v2-8
+      accelerator_args:
+         runtime_version: 2.5.0 # TPU software version to be used.
+
+   # TPU node requires loading data from a GCS bucket.
+   file_mounts:
+      /dataset:
+         name: mnist-tpu-node
+         store: gcs
+         mode: MOUNT
+
+   # The setup command.  Will be run under the working directory.
+   setup: |
+      git clone https://github.com/tensorflow/models.git
+
+      conda activate mnist
+      if [ $? -eq 0 ]; then
+         echo 'conda env exists'
+      else
+         conda create -n mnist python=3.8 -y
+         conda activate mnist
+         pip install tensorflow==2.5.0 tensorflow-datasets tensorflow-model-optimization cloud-tpu-client
+      fi
+
+   # The command to run.  Will be run under the working directory.
+   run: |
+      conda activate mnist
+      cd models/official/legacy/image_classification/
+
+      export STORAGE_BUCKET=gs://mnist-tpu-node
+      export MODEL_DIR=${STORAGE_BUCKET}/mnist
+      export DATA_DIR=${STORAGE_BUCKET}/data
+
+      export PYTHONPATH=/home/gcpuser/sky_workdir/models
+
+      python3 mnist_main.py \
+         --tpu=${TPU_NAME} \
+         --model_dir=${MODEL_DIR} \
+         --data_dir=${DATA_DIR} \
+         --train_epochs=10 \
+         --distribution_strategy=tpu \
+         --download
+
+.. note::
+
+   TPU node requires loading data from a GCS bucket, so we add a :code:`file_mounts` to create a new bucket.
+   Check :ref:`SkyPilot Storage <sky-storage>` for more details.
+
+With the above YAML, you should be able to launch the training job with :code:`sky launch`!
+
+.. code-block:: console
+
+   $ sky launch mnist-tpu-node.yaml -c mycluster
+   ...
+   (tpu_app pid=28961) Epoch 9/10
+   (tpu_app pid=28961) 58/58 [==============================] - 1s 19ms/step - loss: 0.1181 - sparse_categorical_accuracy: 0.9646 - val_loss: 0.0921 - val_sparse_categorical_accuracy: 0.9719
+   (tpu_app pid=28961) Epoch 10/10
+   (tpu_app pid=28961) 58/58 [==============================] - 1s 20ms/step - loss: 0.1139 - sparse_categorical_accuracy: 0.9655 - val_loss: 0.0831 - val_sparse_categorical_accuracy: 0.9742
+   ...
+   (tpu_app pid=28961) {'accuracy_top_1': 0.9741753339767456, 'eval_loss': 0.0831054300069809, 'loss': 0.11388632655143738, 'training_accuracy_top_1': 0.9654667377471924}
```

Comment:
Here the task name is `tpu_app`, not the same as L58?

Reply:
Ah fixed. thanks for catching this.